polkadot/xcm/xcm-executor/src/traits/event_emitter.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use xcm::latest::prelude::*;
+
+pub trait EventEmitter {
+    fn emit_sent_event(
```

Comment:
The `emit_sent_event` function, with `Self::deposit_event(Event::Sent`, makes it a single-purpose solution.  

I would prefer a more general approach. For example, there’s a very relevant [issue](https://github.com/paritytech/polkadot-sdk/issues/2780) where we want/need to emit a failure event from `ProcessXcmMessage`.  

If we could address this issue using an `EventEmitter`, that would be fantastic.

Reply:
one option is this trait to have multiple functions, each specific to xcm-executor events

`fn emit_sent_event` is one
`fn emit_process_event` is another
etc